html/webappapis/scripting/event-loops/shared-between-contexts.html

```diff
@@ -0,0 +1,56 @@
+<!DOCTYPE html>
+<head>
+<meta charset=utf-8>
+<link rel=help href="https://html.spec.whatwg.org/multipage/webappapis.html#event-loops">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+</head>
+<body>
+  <script>
+// 8.1.4.1 Definitions
+//
+// > [...]
+// >
+// > There must be at least one browsing context event loop per user agent, and
+// > at most one per unit of related similar-origin browsing contexts.
+
+async_test(function(t) {
+    new Promise(function(resolve) {
+        var child = window.open('/common/blank.html');
+        child.onload = function() {
+            resolve(child);
+          };
+      })
+      .then(function(child) {
+          var fulfilled = child.Promise.resolve();
+
+          child.close();
+
+          Promise.resolve(fulfilled).then(function() {
```

Comment:
Short answer: because Firefox passes when using `fulfilled.then`.

My guess here is that it comes down to [PerformPromiseThen](https://tc39.github.io/ecma262/#sec-performpromisethen). Firefox seems like it is correctly using the event loop when it comes to calling EnqueueJob from `fulfulled`, but that appending to the [[PromiseFulfillReactions]] of a newly-created Promise is where things go wrong.

Maybe it would be best to include tests for both cases (so four tests total). Does that sound good to you?